[IMP] Inventory: rewrite DHL integration (#10836)
odoo/documentation: content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/dhl_credentials.rst

least one *shipping method*, the process of :doc:`calculating shipping rates
<../setup_configuration>` and :doc:`generating shipping labels <labels>` is greatly simplified.

.. note::
   While a variety of shipping connectors are available for different carriers, this documentation
   details the configuration settings specific to integrating DHL. For instructions on configuring
   the integration settings common to all shippers, see the documentation on :doc:`third-party
   shippers <third_party_shipper>`.

Enable DHL shipping connector
=============================

Before creating a DHL shipping method, it is necessary to enable the carrier's shipping connector.
To do so, navigate to :menuselection:`Inventory app --> Configuration --> Settings`.

Scroll down to the :guilabel:`Shipping Connectors` section, and tick the checkbox next to
:guilabel:`DHL Express Connector`. Finally, click :guilabel:`Save` to apply the changes.

Once the connector has been enabled, a :icon:`oi-arrow-right` :guilabel:`DHL Shipping Methods` link
appears below. Click the link to open a page showing all shipping methods configured for DHL.

> .. image:: dhl_credentials/dsm.png
>    :align: center
>    :alt: The DHL Shipping Methods button on the Settings page of the Inventory app.

Reviewer: Only suggestion is to remove this image, as it is not entirely necessary IMO :)
```suggestion
```

Configure DHL shipping method
=============================

After enabling the shipping connector for DHL, shipping methods can be configured for the carrier.
Once configured, a shipping method can be added as a line item to sales orders (SOs), which allows
for automatic computation of shipping rates, and generation of shipping labels.

To create a new DHL shipping method, navigate to :menuselection:`Inventory app --> Configuration -->
Settings`. In the :guilabel:`Shipping Connectors` section, select the :guilabel:`DHL Shipping
Methods` button below the :guilabel:`DHL Express Connector` checkbox.

.. note::
   It is also possible to see existing shipping methods for every carrier, by navigating to
   :menuselection:`Inventory app --> Configuration --> Shipping Methods`.

Click :guilabel:`New` to open a blank shipping method form. If a shipping method has already been
created, it can be selected from this screen.

.. image:: dhl_credentials/dhl-form.png
   :align: center
   :alt: The form for a DHL shipping method.

General information
-------------------

Begin configuring the shipping method by entering its title in the :guilabel:`Shipping Method`
field.

> 

Author: That one is general to all shippers, so it's not covered in this doc.

Developer Portal <https://developer.dhl.com/user/register>`_, by filling out the following
information:

- :guilabel:`First name`
- :guilabel:`Last name`
- :guilabel:`Email address`
- :guilabel:`Username`
- :guilabel:`Company Name`
- :guilabel:`Country`

After filling out all of the fields, tick the :guilabel:`I have read and accept the terms of use and
the privacy policy` checkbox, and click :guilabel:`Send` to register.

After registering, a confirmation email is sent to verify the application. Click the link inside the
email to activate the developer portal account, and set a password.

This email also includes the account number for the developer portal account. Enter the number in
the :guilabel:`DHL Account Number` field on the shipping method form.

Once the developer portal account has been confirmed, log in to the portal using the username and
password. Click on the user avatar in the top-right corner of the screen to open the user dashboard.

On the dashboard, click on the :guilabel:`Apps` tab, and select an app. In the
:guilabel:`Credentials` section, click the :guilabel:`Show key` button below the :guilabel:`API Key`
and :guilabel:`API Secret` fields.

> Return to the shipping method form in the Odoo database. Enter the :guilabel:`DHL API Key` in the
> :guilabel:`SiteID` field, and the :guilabel:`API Secret` in the :guilabel:`DHL Password` field.

Reviewer 2: ```suggestion
Return to the shipping method form in the Odoo database. Enter the *DHL API Key* in the
:guilabel:`DHL SiteID` field, and the *API Secret* in the :guilabel:`DHL Password` field.
```

Shipping details
~~~~~~~~~~~~~~~~

The rest of the fields in the :guilabel:`DHL Configuration` tab are used to configure the shipping
method itself:

- :guilabel:`Region`: the region in which the shipping method is used.

> - :guilabel:`DHL Product`: the shipping service purchased from DHL (ex. Express Worldwide).

Reviewer 2: ```suggestion
- :guilabel:`DHL Product`: the shipping service purchased from DHL (e.g. Express Worldwide).
```

> - :guilabel:`DHL Package Type`: the type of DHL package used for delivery (ex. DHL Box).

Reviewer 2: ```suggestion
- :guilabel:`DHL Package Type`: the type of DHL package used for delivery (e.g. DHL Box).
```

- :guilabel:`Package Weight Unit`: the unit of measure used to display package weight.
- :guilabel:`Package Dimension Unit`: the unit of measure used to display package size.
- :guilabel:`Label Format`: the file format used to generate shipping labels.
- :guilabel:`Label Template`: the paper size used to print shipping labels.

.. important::

>    Before selecting service options for a shipping method, make sure that those services are

Reviewer 2: ```suggestion
   Before selecting service options for a shipping method, make sure those services are
```

   actually available for the DHL account. Available services depend on the contract negotiated with
   DHL.

Options
~~~~~~~

> A couple of additional settings are available in the :guilabel:`Options` section at the bottom of

Reviewer 2: ```suggestion
Additional settings are available in the :guilabel:`Options` section at the bottom of
```

the :guilabel:`DHL Configuration` tab:

> - :guilabel:`Generate Return Label`: enable this option to automatically generate a return label

Reviewer 2: ```suggestion
- :guilabel:`Generate Return Label`: Enable this option to automatically generate a return label
```

  after validating a delivery order.

> - :guilabel:`Dutiable Material`: enable this option if the shipping method is liable to customs or

Reviewer 2: ```suggestion
- :guilabel:`Dutiable Material`: Enable this option if the shipping method is liable to customs or
```
